src/components/Composer/index.native.js

```diff
@@ -0,0 +1,168 @@
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            propStyles: StyleSheet.flatten(this.props.style),
+        };
+    }
+
+    componentDidMount() {
+        // This callback prop is used by the parent component using the constructor to
+        // get a ref to the inner textInput element e.g. if we do
+        // <constructor ref={el => this.textInput = el} /> this will not
+        // return a ref to the component, but rather the HTML element by default
+        if (!this.props.forwardedRef || !_.isFunction(this.props.forwardedRef)) {
+            return;
+        }
+
+        this.props.forwardedRef(this.textInput);
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.shouldClear || !this.props.shouldClear) {
+            return;
+        }
+
+        this.textInput.clear();
+        this.props.onClear();
+    }
+
+    /**
+     * Calculates the max number of lines the text input can have
+     *
+     * @param {Number} lineHeight
+     * @param {Number} paddingTopAndBottom
+     * @param {Number} scrollHeight
+     *
+     * @returns {Number}
+     */
+    getNumberOfLines(lineHeight, paddingTopAndBottom, scrollHeight) {
+        return Math.ceil((scrollHeight - paddingTopAndBottom) / lineHeight);
+    }
+
+    /**
+     * Check the current scrollHeight of the textarea (minus any padding) and
+     * divide by line height to get the total number of rows for the textarea.
+     * @param {Event} e
+     */
+    updateNumberOfLines(e) {
+        const lineHeight = this.state.propStyles.lineHeight;
+        const paddingTopAndBottom = styles.textInputComposeSpacing.paddingVertical * 2;
+        const inputHeight = lodashGet(e, 'nativeEvent.contentSize.height', null);
+        if (!inputHeight) {
+            return;
+        }
+        const numberOfLines = this.getNumberOfLines(lineHeight, paddingTopAndBottom, inputHeight);
+        this.updateIsFullComposerAvailable(numberOfLines);
+    }
+
+    /**
+     * Update isFullComposerAvailable if needed
+     * @param {Number} numberOfLines The number of lines in the text input
+     */
+    updateIsFullComposerAvailable(numberOfLines) {
+        let isFullComposerAvailable = false;
+        if (numberOfLines >= CONST.REPORT.FULL_COMPOSER_MIN_LINES) {
+            isFullComposerAvailable = true;
+        }
+        if (isFullComposerAvailable !== this.props.isFullComposerAvailable) {
+            this.props.setIsFullComposerAvailable(isFullComposerAvailable);
+        }
+    }
+
+    render() {
+        // On native layers we like to have the Text Input not focused so the user can read new chats without they keyboard in
+        // the way of the view.
+        // On Android, the selection prop is required on the TextInput but this prop has issues on IOS
+        // https://github.com/facebook/react-native/issues/29063
+        const propsToPass = Platform.OS === 'ios' ? _.omit(this.props, 'selection') : this.props;
```

Comment:
I think we try to stay away from `Platform.OS`.

Reply:
I think `Platform.OS` is worth it here, because otherwise I would have a complete duplicate file where the only difference is not passing this prop. I decided to combine the iOS and Android implementations since I would be adding more functionality to this class.

If you feel strongly that this is a bad approach then I could refactor so that we once again have separate implementations, and I can try to extract the shared functions into a utility file.

Reply:
Yea, it is unfortunate that there would be so much overlap... 
Perhaps there's a different way to accomplish this.

It's not really a matter of feeling strongly or not. We just don't use Platform.OS as a matter of policy.

Reply:
Ok. I am creating separate files for ios and android again. I'm moving `updateNumberOfLines` and `getNumberOfLines` into `ComposerUtils/index.native.js` for the mobile implementations and I'll move the web / desktop implementations of those methods into `ComposerUtils/index.js`.

Reply:
Yup! I think that's a great model for how to do this. Thanks @rushatgabhane

Reply:
The `BaseModal` holds shared functionality for all platforms. In this case, native is quite different from web so I think the way I refactored makes sense. Please let me know what you think of my latest changes.